django_restql/mixins.py

```diff
@@ -552,4 +553,147 @@ def update(self, instance, validated_data):
+                values.extend(self.get_all_dict_values(value))
+            else:
+                values.append(value)
+
+        return values
+
+    def get_mapping_values(self, parsed, mapping):
+        """
+        Returns the mapping value (or nested mapping values as needed) of a particular parsed dict
+        against the mapping provided. Parsed input expected to come from get_dict from the parser.
+        """
+        values = []
+
+        for parsed_key, parsed_value in parsed.items():
+            if parsed_key in mapping.keys():
+                mapping_value = mapping[parsed_key]
+
+                if isinstance(mapping_value, dict):
+                    base = mapping_value.get("base")
+                    nested = mapping_value.get("nested")
+                else:
+                    base = mapping_value
+                    nested = None
+
+                # This should never be a falsy value, but we're being safe here.
+                if base:
+                    values.append(base)
+
+                if nested:
+                    # If we're given a dict, we only want keys that were mapped as needed, so
+                    # recursively call with the smaller map.
+                    if isinstance(parsed_value, dict):
+                        nested_values = self.get_mapping_values(parsed_value, nested)
+                        values.extend(nested_values)
+                    else:
+                        # If we don't have a dict, we want every nested value, since it's assumed
+                        # all of them will be present. We recursively get all values from here.
+                        nested_values = self.get_all_dict_values(nested)
+                        values.extend(nested_values)
+        return values
+
+    def apply_restql_orm_mapping(self, queryset):
+        """
+        Applies appropriate select_related and prefetch_related calls on a
+        queryset based on the passed on dictionaries provided.
+        """
+        parsed_keys = self.get_restql_query_dict()
+        select = self.get_select_related_mapping()
+        prefetch = self.get_prefetch_related_mapping()
+
+        select_mapped = self.get_mapping_values(parsed_keys, select)
+        prefetch_mapped = self.get_mapping_values(parsed_keys, prefetch)
+
+        for value in select_mapped:
+            if isinstance(value, str):
+                queryset = queryset.select_related(value)
+            elif isinstance(value, list):
+                for select_value in value:
+                    queryset = queryset.select_related(select_value)
+
+        for value in prefetch_mapped:
+            if isinstance(value, str) or isinstance(value, Prefetch):
+                queryset = queryset.prefetch_related(value)
+            elif isinstance(value, list):
+                for prefetch_value in value:
+                    queryset = queryset.prefetch_related(prefetch_value)
+
+        return queryset
```

Comment:
I think the name of this method should be `get_optimized_queryset` because it returns a performance optimized queryset.

Reply:
I am hesitant to have the word `optimized` etc used here (not a blocker, just my opinion). This is not a "better" queryset optimizer than django's ORM but instead the built-in tooling being applied in an API friendly scheme.

`setup_eager_loading` could be another option (admittedly it's not `eager` so much as its planned ahead of time, but that's too verbose!). The article is a bit old, but curious to have thoughts on it (and it's serializer based, but that's not a big deal). https://ses4j.github.io/2015/11/23/optimizing-slow-django-rest-framework-performance/

Reply:
That's brilliant, I read that article few months ago, that name makes more sense than the other two, now that you have mentioned it I think it would be great to rename `RestQLViewMixin` to `EagerLoadingMixin`.

Reply:
Went with `apply_eager_loading`, since this is actively applying our eager loading to the provided queryset.